app/client/ts/spec/user-profile-spec.ts

```diff
@@ -0,0 +1,50 @@
+import { UserProfileData, UserProfile } from '../user-model';
+import { safeClient } from '../util';
+import { sleep, makeAlphaid, failDone } from './test-util';
+import Config from '../global-config';
+
+const CONFIG: Config = Config.getInstance();
+
+let VALID_LONG_NAME: string;
+const HOME_DIR: string = makeAlphaid().toLowerCase();
+
+describe('The user profile data store', () => {
+    it('can detect when no profile exists', async (done) => {
+        try {
+            const profile: UserProfile = await UserProfile.read('does-not-exist');
+        } catch (e) {
+            done();
+        }
+        fail();
+    });
+
+    beforeAll(async (done) => {
+        VALID_LONG_NAME = makeAlphaid().toLowerCase();
+
+        failDone(safeClient.nfs.dir.create('app', HOME_DIR, false), done);
```

Comment:
I would probably `await` these guys just to make sure that they get run in the right order.